error deleting a metal_vlan resource if the vlan is still associated with a connection

When I try to delete a **metal_vlan** resource and it is still associated with a connection it returns below panic error instead of a handled error message. The error is fixed after manually removing the association to the connection in the portal, maybe this information should be included in an error message

**$ terraform -version**
Terraform v0.15.0
on linux_amd64
+ provider registry.terraform.io/equinix/equinix v1.1.0
+ provider registry.terraform.io/equinix/metal v2.0.1

**Console output**
metal_vlan.vlan: Destroying... [id=3ce69483-8837-4077-84fb-e92d1b119514]
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-metal_v2.0.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xccca40]

goroutine 25 [running]:
github.com/equinix/terraform-provider-metal/metal.ignoreResponseErrors.func1(0x0, 0xfef720, 0xc0004ed2f0, 0xc0006a0c60, 0xfef720)
	github.com/equinix/terraform-provider-metal/metal/errors.go:144 +0x60
github.com/equinix/terraform-provider-metal/metal.resourceMetalVlanDelete(0xc000221500, 0xdcd380, 0xc0003e0500, 0x24, 0x150e860)
	github.com/equinix/terraform-provider-metal/metal/resource_metal_vlan.go:129 +0x12c
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000426180, 0xc00020bc20, 0xc000667d40, 0xdcd380, 0xc0003e0500, 0x40e001, 0x20, 0xe3e580)
	github.com/hashicorp/terraform-plugin-sdk@v1.0.0/helper/schema/resource.go:278 +0x4a6
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000426300, 0xc0002d98e8, 0xc00020bc20, 0xc000667d40, 0x150e860, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk@v1.0.0/helper/schema/provider.go:289 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000eb28, 0x10012d0, 0xc00021dd40, 0xc00020f2c0, 0xc00000eb28, 0xc00021dd40, 0xc000754a50)
	github.com/hashicorp/terraform-plugin-sdk@v1.0.0/internal/helper/plugin/grpc_provider.go:885 +0x88b
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xe691c0, 0xc00000eb28, 0x10012d0, 0xc00021dd40, 0xc00020f260, 0x0, 0x10012d0, 0xc00021dd40, 0xc0000bc620, 0xd6)
	github.com/hashicorp/terraform-plugin-sdk@v1.0.0/internal/tfplugin5/tfplugin5.pb.go:3189 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000be000, 0x10092f8, 0xc00073c900, 0xc000424d00, 0xc00041cf30, 0x14cfd60, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.23.0/server.go:995 +0x482
google.golang.org/grpc.(*Server).handleStream(0xc0000be000, 0x10092f8, 0xc00073c900, 0xc000424d00, 0x0)
	google.golang.org/grpc@v1.23.0/server.go:1275 +0xd2c
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000366220, 0xc0000be000, 0x10092f8, 0xc00073c900, 0xc000424d00)
	google.golang.org/grpc@v1.23.0/server.go:710 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.23.0/server.go:708 +0xa5

Error: The terraform-provider-metal_v2.0.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/metal/errors.go
+++ b/metal/errors.go
@@ -140,6 +140,9 @@ func httpNotFound(resp *http.Response, err error) bool {
 // provided checks
 func ignoreResponseErrors(ignore ...func(resp *http.Response, err error) bool) func(resp *packngo.Response, err error) error {
 	return func(resp *packngo.Response, err error) error {
+		if resp == nil || resp.Response == nil {
+			return err
+		}
 		for _, ignored := range ignore {
 			if !ignored(resp.Response, err) {
 				return err

